modules/ppcp-wc-gateway/src/Gateway/PayUponInvoice/PayUponInvoice.php

```diff
@@ -342,6 +342,43 @@ function () {
 				}
 			}
 		);
+
+		add_action(
+			'woocommerce_update_options_checkout_ppcp-pay-upon-invoice-gateway',
+			function () {
+				$customer_service_instructions = filter_input( INPUT_POST, 'woocommerce_ppcp-pay-upon-invoice-gateway_customer_service_instructions', FILTER_SANITIZE_STRING );
+				if ( '' === $customer_service_instructions ) {
+					$gateway_settings = get_option( 'woocommerce_ppcp-pay-upon-invoice-gateway_settings' );
```

Comment:
Would be great to at least extract this name somewhere.

Or a simple wrapper `WcGatewaySettings` with functions like `load(): array`, `save(array)`. Or a `ContainerInterface` wrapper like [Settings](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-wc-gateway/src/Settings/Settings.php).